jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -256,75 +267,45 @@ public boolean isContainedIn(Resource r)
     @Override
     public URI getTargetURI()
     {
-        if (!targetResolved)
-        {
-            targetPath = resolveTargetPath();
-            targetResolved = true;
-        }
-        if (targetPath == null)
-            return null;
+        checkAlias();
         return targetPath.toUri();
     }
 
-    private Path resolveTargetPath()
+    private void checkAlias()
     {
-        Path abs = path;
-
-        // TODO: is this a valid shortcut?
-        // If the path doesn't exist, then there's no alias to reference
-        if (!Files.exists(path))
-            return null;
-
-        /* Catch situation where the Path class has already normalized
-         * the URI eg. input path "aa./foo.txt"
-         * from an #resolve(String) is normalized away during
-         * the creation of a Path object reference.
-         * If the URI is different from the Path.toUri() then
-         * we will just use the original URI to construct the
-         * alias reference Path.
-         *
-         * We use the method `toUri(Path)` here, instead of
-         * Path.toUri() to ensure that the path contains
-         * a trailing slash if it's a directory, (something
-         * not all FileSystems seem to support)
-         */
-        if (!URIUtil.equalsIgnoreEncodings(uri, toUri(path)))
+        if (targetPath == null)
```

Comment:
The targetPath can sometimes not be obtained.
FileSystem issues, IO issues, etc.
So this could be null.

Reply:
And we don't want to resolve again, so we need a special object to an additional boolean.

Reply:
If you look at my implementation of `resolveTargetPath()` you'll see that it never returns null.
It only attempts to resolve aliases and if it can't be done it will return the same path it was given.

So the `targetPath` field acts as the flag to determine whether we have checked for aliases.

Reply:
I think we (sbordet and I) both think `resolveTargetPath()` shouldn't return a Path on failure